plasmapy/diagnostics/path_integrated_diagnostic.py

```diff
@@ -0,0 +1,627 @@
+            detector horizontal axis.
+
+        verbose : bool, optional
+            If true, updates on the status of the program will be printed
+            into the standard output while running.
+
+        """
+
+        # Validate the quantities input
+        if isinstance(quantities, str):
+            quantities = [
+                quantities,
+            ]
+
+        for quantity in quantities:
+            if quantity not in grid.quantities:
+                raise ValueError(
+                    f"quantity {quantity} is not defined on the " "provided grid."
+                )
+
+        self.quantities = quantities
+
+        # Continue with the rest of the parent class init
+        super().__init__(grid, source, detector, verbose=verbose)
+
+    def _integrand(self, pts):
+        r"""
+        Returns the scalar value of the quantity or quantities specified
+        at each point.
+
+        Parameters
+        ----------
+
+        pts: `~astropy.units.Quantity` (nx*ny*nz, 3)
+            Positions at which the integrand will be evaluated.
+
+        Returns
+        -------
+
+        arr: `~astropy.units.Quantity` (nx*ny*nz) or list of same
+            Integrand value at each of the points provided. Some integrand
+            functions may return multiple arrays of interpolated values as
+            a list, each of which will then be integrated separately.
+
+        """
+        # Reshape the pts array from grid shape (nx, ny, nz, 3) to a list
+        # of points (nx*ny*nz, 3) as required by the grids interpolators
+        nx, ny, nz, ndim = pts.shape
+        pts = np.reshape(pts, (nx * ny * nz, ndim))
+
+        # If volume_averaged_interpolator not defined for this grid
+        # (eg. for non-uniform grids)
+        # use the nearest neighbor interpolator instead
+        try:
+            integrand = self.grid.volume_averaged_interpolator(pts, *self.quantities)
+        except NotImplementedError:
+            integrand = self.grid.nearest_neighbor_interpolator(pts, *self.quantities)
+
+        # Reshape the integrands from (nx*ny*nz) to (nx, ny, nz)
+        integrand = np.reshape(integrand, (nx, ny, nz))
+
+        return integrand
+
+    def evaluate(
+        self,
+        num,
+        size=None,
+        bins=[200, 200],
+        collimated=True,
+    ):
+
+        """
+        Evaluates the line integral through the
+        provided grid. This is accomplished through the following steps:
+
+        1) Create an array of points in the detector plane using the detector
+        location and the size and bins keywords.
+
+        2) For each cell of detector grid, create an array of points of
+        separation ds from there to the source point (or, when collimated=True,
+        the source plane).
+
+        3) Evaluate the integrand function at each point.
+
+        4) Integrate along the line-integrated dimension to obtain the
+        line-integrated quantity in the detector plane.
```

Comment:
This is probably causing the new errors since the line continuation is suppose to indent.